### ISSUE TITLE
KoBoForm should warn users when deleting questions referenced by other questions

### ISSUE THREAD

**Issue by Dylan-Gillespie** [kobotoolbox/koboform#18]
_From [j...@ethertontech.com](https://code.google.com/u/102366118286685495754/) on April 27, 2012 17:26:24_

Summary: When deleting a question that is referenced by other questions for valids or relevants, KoBoForm should warn a user that deleting the question will affect the functionality of other questions proposed use-case: User clicks delete on a question referenced by another question and KoBoForm warns them that another question is using the question. The warning should let the user know both the name of the question referencing, and being referenced. additional comments:

_Original issue: http://code.google.com/p/kobo/issues/detail?id=18_

**Comment by Dylan-Gillespie** [comment]
_From [j...@ethertontech.com](https://code.google.com/u/102366118286685495754/) on April 27, 2012 23:28:13_

Neil can you test this.

**Status:** Fixed  
  **Cc:** neil.hen...@kobotoolbox.org

**Comment by Dylan-Gillespie** [comment]
_From [neil.hen...@kobotoolbox.org](https://code.google.com/u/109155374072957516293/) on May 03, 2012 10:38:51_

Great. 
I am testing, BTW, in the BETA formbuilder, not in the main site one. (a mistake I made for a while, thinking this wasn't working.)
Anyway, I tested it and it's working fine. I get the following error when I try to delete a question that is referred to in subsequent question's skip logic:

Warning question"Favorite Color" is referenced by the following questions as a skip logic condtion: "Really, Red?", "Really, Blue?", "Really, Green?".

Are you sure you want to delete question "Favorite Color"

**Status:** Verified  

**Attachment:** [Favorite Color.xml.xml](http://code.google.com/p/kobo/issues/detail?id=18#hc2)
